### PR TITLE
[NFC] Replace uses of find(x) != end() idiom with contains(x) for StringRef and Set-like types

### PR DESCRIPTION
**Pull request by mininny**
> A little while back, LLVM Set-like data types got contains(t) methods which make code more readable. Many places in the compiler use the find(a) != .end() idiom to check for membership (and in some places find(a) == .end() to check if the item is not present).
>
> It would be nice to update the code to use contains(a) to improve the readability.

This PR replaces usages of `.find(Key) != .end()` idiom with `.contains(Key)`.

Resolves [SR-13950.](https://bugs.swift.org/browse/SR-13950)

@varungandhi-apple Could you please review this? :) 🙏

### CODE REVIEW

File: unittests/runtime/Mutex.cpp [hunk 1]
```diff
@@ -856,11 +856,12 @@ template <typename RW> void readWriteLockCacheExampleThreaded(RW &lock) {
...
+          } else {
+            if (trace)
+              printf("Worker[%d] HIT for key = %d, value = %d.\n", i, key,
+                     value->second);
+            found = true; // cache hit, no need to grab write lock
           }
-          if (trace)
-            printf("Worker[%d] HIT for key = %d, value = %d.\n", i, key,
-                   value->second);
-          found = true; // cache hit, no need to grab write lock
```
Comment (mininny) [inline unittests/runtime/Mutex.cpp:863]: I also added the missing `else` in this code, but I'm not sure if it is okay to fix it in this PR as well?

File: lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -192,13 +192,13 @@ static void validateDebugInfoArgs(DiagnosticEngine &diags,
...
   for (auto A : args.getAllArgValues(options::OPT_debug_prefix_map))
-    if (A.find('=') == StringRef::npos)
+    if (!A.contains('=')))
       diags.diagnose(SourceLoc(), diag::error_invalid_debug_prefix_map, A);
 
   // Check for any -coverage-prefix-map options that aren't of the form
   // 'original=remapped' (either side can be empty, however).
   for (auto A : args.getAllArgValues(options::OPT_coverage_prefix_map))
-    if (A.find('=') == StringRef::npos)
+    if (!(A.contains('=')))
```
Comment (gottesmm) [inline lib/Driver/Driver.cpp:201]: I don't think you need the () here.

File: lib/Driver/Driver.cpp [hunk 3]
```diff
@@ -661,7 +661,7 @@ static SmallVector<StringRef, 8> findRemovedInputs(
   SmallVector<StringRef, 8> missingInputs;
   for (auto &previousInput : previousInputs) {
     auto previousInputArg = previousInput.getKey();
-    if (inputArgs.find(previousInputArg) == inputArgs.end()) {
+    if (!(inputArgs.contains(previousInputArg))) {
```
Comment (gottesmm) [inline lib/Driver/Driver.cpp:664]: same here.

File: lib/Frontend/DiagnosticVerifier.cpp [hunk 4]
```diff
@@ -128,7 +128,7 @@ findDiagnostic(std::vector<CapturedDiagnosticInfo> &CapturedDiagnostics,
 
     // Verify the classification and string.
     if (I->Classification != Expected.Classification ||
-        I->Message.find(Expected.MessageStr) == StringRef::npos)
+        !(I->Message.contains(Expected.MessageStr)))
```
Comment (gottesmm) [inline lib/Frontend/DiagnosticVerifier.cpp:131]: in fact in most of these places, I don't think you need the extra () for the !. The ! should bind tight I think.

File: lib/LLVMPasses/ARCEntryPointBuilder.h [hunk 5]
```diff
@@ -193,8 +193,7 @@ class ARCEntryPointBuilder {
     // intrinsics are atomic today.
     if (I->getIntrinsicID() != llvm::Intrinsic::not_intrinsic)
       return false;
-    return (I->getCalledFunction()->getName().find("nonatomic") !=
-            llvm::StringRef::npos);
+    return (I->getCalledFunction()->getName().contains("nonatomic"));
```
Comment (gottesmm) [inline lib/LLVMPasses/ARCEntryPointBuilder.h:196]: It looks like you are handling StringRef and Set-like types. Really this should be two separate commits. I am fine if it is in the same PR. But it should be explicit since a StringRef is /not/ a set like type and contains in the context of a StringRef means something else.

File: lib/SILOptimizer/Analysis/ARCAnalysis.cpp [hunk 6]
```diff
@@ -964,7 +963,7 @@ void ConsumedArgToEpilogueReleaseMatcher::collectMatchingReleases(
     if (!isa<DestroyAddrInst>(inst)) {
       // We do not know if we have a fully post dominating release
       // set, so we use the partial post dom entry point.
-      if (auto partialReleases = iter->second.getPartiallyPostDomReleases()) {
+      if (auto partialReleases = ArgInstMap[arg].getPartiallyPostDomReleases()) {
```
Comment (gottesmm) [inline lib/SILOptimizer/Analysis/ARCAnalysis.cpp:966]: We don't want to make this change. We are in this case doing two different look ups. We want to do a single lookup.

File: lib/SILOptimizer/Analysis/ARCAnalysis.cpp [hunk 7]
```diff
@@ -973,7 +972,7 @@ void ConsumedArgToEpilogueReleaseMatcher::collectMatchingReleases(
 
     // We've seen part of this base, but this is a part we've have not seen.
     // Record it.
-    iter->second.addRelease(&inst);
+    ArgInstMap[arg].addRelease(&inst);
```
Comment (gottesmm) [inline lib/SILOptimizer/Analysis/ARCAnalysis.cpp:975]: same here. Changing something like that should be done in a separate commit.

**Review by gottesmm — COMMENTED** [review]
A few thoughts:

1. You are not only changing set like things here. In fact, it makes me worried in general about this patch. Instead you are changing some sort like things and in some cases StringRef. Please separate the StringRef changes into a separate commit.
2. I found at least 1-2 cases where the patch is changing a single lookup into multiple lookups. Please remove that (I marked it).

Before this goes in I want to look at it again.

File: lib/Frontend/DiagnosticVerifier.cpp (hunk 4 above)
Comment (mininny) [reply lib/Frontend/DiagnosticVerifier.cpp]: Thank you! I removed the extra () for ! everywhere :)

File: lib/LLVMPasses/ARCEntryPointBuilder.h (hunk 5 above)
Comment (mininny) [reply lib/LLVMPasses/ARCEntryPointBuilder.h:196]: Done!

File: lib/SILOptimizer/Analysis/ARCAnalysis.cpp (hunk 6 above)
Comment (mininny) [reply lib/SILOptimizer/Analysis/ARCAnalysis.cpp]: Thanks, I reverted changes to places where # of look ups are significant.

File: lib/SILOptimizer/PassManager/PassManager.cpp [hunk 8]
```diff
@@ -233,8 +230,7 @@ static void printModule(SILModule *Mod, bool EmitVerboseSIL) {
         std::find(SILPrintOnlyFun.begin(), SILPrintOnlyFun.end(), F.getName()))
       F.dump(EmitVerboseSIL);
 
-    if (!SILPrintOnlyFuns.empty() &&
-        F.getName().find(SILPrintOnlyFuns, 0) != StringRef::npos)
```
Comment (gottesmm) [inline lib/SILOptimizer/PassManager/PassManager.cpp:237]: A note for myself. Whats with the 0?

Comment (gottesmm) [reply lib/SILOptimizer/PassManager/PassManager.cpp:237]: @mininny I looked in StringRef.h. Looks like the 0 is a default argument to find that says start checking at the beginning. (Just answering my own question).

File: unittests/runtime/Mutex.cpp (hunk 1 above)
Comment (gottesmm) [reply unittests/runtime/Mutex.cpp:863]: I would put it in a different PR. Different high level thing.

**Review by gottesmm — APPROVED** [review]

**Comment by gottesmm** [comment]
Just please move that last commit to a different PR and I think this is ready to go in.

**Comment by gottesmm** [comment]
@swift-ci test

**Comment by gottesmm** [comment]
Doing a quick test to see if everything works.

File: lib/LLVMPasses/LLVMARCOpts.cpp [hunk 9]
```diff
@@ -142,8 +142,8 @@ static bool canonicalizeInputFunction(Function &F, ARCEntryPointBuilder &B,
         // Have not encountered a strong retain/release. keep it in the
         // unknown retain/release list for now. It might get replaced
         // later.
-        if (NativeRefs.find(ArgVal) == NativeRefs.end()) {
-           UnknownObjectRetains[ArgVal].push_back(&CI);
+        if (!NativeRefs.contains(ArgVal)) {
```
Comment (LucianoPAlmeida) [inline lib/LLVMPasses/LLVMARCOpts.cpp:145]: Seems the build is failing because`llvm::DenseSet` doesn't have a count method, @gottesmm we normally use [`count`](http://llvm.org/doxygen/classllvm_1_1detail_1_1DenseSetImpl.html#afe504aa31a6a354cec13f5b32d0b1d9d) so perhaps we could use this here as well? :)

Comment (mininny) [reply lib/LLVMPasses/LLVMARCOpts.cpp:145]: It seems like the change to `llvm::DenseSet` hasn't landed on `swift/main` branch so it's causing this issue..? The `contains` method exists on llvm-project's `apple/stable/20210107` branch; I'm not sure how the CI works but could this be fixed by moving the base branch? :)

Comment (LucianoPAlmeida) [reply lib/LLVMPasses/LLVMARCOpts.cpp:145]: Yeah, seems like it hasn't ... I would say that use `count` instead for `Set` like would just be fine and already an improvement, but since the whole point of the SR is to use `contains` I'm not really sure about that, so cc @varungandhi-apple @gottesmm

**Comment by CodaFi** [comment]
LGTM

@swift-ci test

**Comment by CodaFi** [comment]
Please revert the DenseSet refactoring

```
/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/lib/LLVMPasses/LLVMARCOpts.cpp:145:25: error: no member named 'contains' in 'llvm::DenseSet<llvm::Value *, llvm::DenseMapInfo<llvm::Value *> >'
14:36:13         if (!NativeRefs.contains(ArgVal)) {
14:36:13              ~~~~~~~~~~ ^
14:36:13 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-main/swift/lib/LLVMPasses/LLVMARCOpts.cpp:192:25: error: no member named 'contains' in 'llvm::DenseSet<llvm::Value *, llvm::DenseMapInfo<llvm::Value *> >'
14:36:13         if (!NativeRefs.contains(ArgVal)) {
14:36:13              ~~~~~~~~~~ ^
```

**Comment by varungandhi-apple** [comment]
@CodaFi I posted this in a thread

> We can wait for a couple of weeks; I'll ping this PR when we have the rebranch so we can run tests and land the change. I don't think cherry-picking changes onto other branches is the right fix here, as that might lead to more merge conflicts.

Reverting is going to be more work, instead we can wait it out for changes to land on the LLVM side.

**Comment by varungandhi-apple** [comment]
Mishal just posted information on the rebranch in the forums: https://forums.swift.org/t/updating-llvm-project-swift-main-branch/44019. We can land this change once the rebranch is done.

**Comment by varungandhi-apple** [comment]
@mininny can you resolve the merge conflicts? I can run the tests after that.

**Comment by varungandhi-apple** [comment]
I think it's okay for now, but for the future, please prefer to rebase for resolving conflicts instead of merging, so that we have a cleaner history.

@swift-ci smoke test

**Comment by mininny** [comment]
> I think it's okay for now, but for the future, please prefer to rebase for resolving conflicts instead of merging, so that we have a cleaner history.
> 
> @swift-ci smoke test

Oh didn't realize that! Okay, Thanks! :)

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Comment by varungandhi-apple** [comment]
Strange, it looks like the change didn't make it in? 🤔

```
/Users/buildnode/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/swift/lib/Frontend/DiagnosticVerifier.cpp:131:21: error: no member named 'contains' in 'llvm::SmallString<128>'
        !I->Message.contains(Expected.MessageStr))
         ~~~~~~~~~~ ^
```

File: lib/Frontend/DiagnosticVerifier.cpp [hunk 10]
```diff
@@ -128,7 +128,7 @@ findDiagnostic(std::vector<CapturedDiagnosticInfo> &CapturedDiagnostics,
 
     // Verify the classification and string.
     if (I->Classification != Expected.Classification ||
-        I->Message.find(Expected.MessageStr) == StringRef::npos)
+        !I->Message.contains(Expected.MessageStr))
```
Comment (varungandhi-apple) [inline lib/Frontend/DiagnosticVerifier.cpp:131]: I think this one and similar checks on `StringRef` are not right. In the JIRA, I only noted the change for Set-like types, I don't think there is a similar method on `StringRef`, which is why there is a CI error here. Did you add a `contains` method on `StringRef` in upstream LLVM?

Comment (mininny) [reply lib/Frontend/DiagnosticVerifier.cpp]: When I was working on this, I found that there was `contains` method in [`StringRef`](https://github.com/apple/llvm-project/blob/1044437905e95a42372d7c1c2d341ebc729570ce/llvm/include/llvm/ADT/StringRef.h#L465) so I worked on them as well. 

I think this failed because `I→Message` turned out to be `SmallString` not `StringRef`. I reverted the change here, could you please run the tests again? :)

**Comment by LucianoPAlmeida** [comment]
@swift-ci smoke test

File: lib/Frontend/DiagnosticVerifier.cpp (hunk 10 above)
Comment (varungandhi-apple) [reply lib/Frontend/DiagnosticVerifier.cpp]: Oh you're right, it's `SmallString` and not `StringRef`. I should not be reviewing code around midnight lol.

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test macOS
